contracts/ControllerCurveStrat.sol

```diff
@@ -71,103 +136,141 @@ contract ControllerCurveStrat is ControllerStratAbs {
         return wantBalance() - _balance;
     }
 
-    function _withdrawFromPool(uint btcCrvAmount) internal {
+    function _withdrawFromPool(uint _wantCrvAmount) internal {
         // Remove staked from gauge
-        ICurveGauge(GAUGE).withdraw(btcCrvAmount);
+        gauge.withdraw(_wantCrvAmount);
 
         // remove_liquidity
-        uint _balance = btcCRVBalance();
-        uint expected = _btcCrvToBtcDoubleCheck(_balance);
+        uint _balance = wantCRVBalance();
+        uint _expected = _wantCrvToWantDoubleCheck(_balance);
 
-        require(expected > 0, "remove_liquidity expected = 0");
+        require(_expected > 0, "remove_liquidity expected = 0");
+
+        if (crvToken.allowance(address(this), pool) == 0) {
```

Comment:
this should be more `if (address(crvToken) != address(pool))` because the allowance **should** be always 0

Reply:
Done! =)